### ISSUE TITLE
Changing Window Location

### ISSUE THREAD

**Issue by Cmpukahi** [Openarl/PathOfBuilding#336]
When I move the window to another monitor, the next time I start up the program, it crashes. The only fix I could find is a fresh install. This resets the location of the window back to the main starting area. I saw that there are problems with GeForce experience causing the program to crash. Another user suggested updating the GeForce Experience, but this didn't help.

Windows 10 Creator build and fully updated GeForce experience.

**Comment by Openarl** [comment]
I haven't been able to make any progress on this one; crashes on startup, including the one caused by GeForce Experience, aren't normally dependant on which monitor the program is running on. What's your exact setup? 

Also, you can avoid the need to reinstall by editing one of the config files, "Launch.cfg", which is with the .exe in the standalone version, and in "%ProgramData%\Path of Building" in the normal installation. You can open it in Notepad and delete everything in it, then save it and start the program; this will reset the window position. BTW, if you do end up having to do that, post the contents of the file here before you delete it, as there might be some hints there as to the cause of the crash.

**Comment by Cmpukahi** [comment]
Sorry for the late response. Another update to windows and GeForce experience got pushed, but doesn't help.

Here's the file contents as requested:
set con_log "0"
set r_compress "0"
set r_screenshotFormat "jpg"
set con_fontSize "14"
set vid_mode "8"
set vid_resizable "3"
set vid_display "-1"
set vid_fullscreen "0"
set vid_last "1920,1017,-1920,23,1"

MSI GTX 1070, dual monitor setup, 16gb RAM, Windows 10 creator update. Using the standalone version right now. Happened on the install one also. I can reproduce it every time by dragging the window to the other monitor. 

For some reason, resizing the window on the main monitor doesn't crash on the next startup.

Here's the file where I resize on the main monitor

set con_log "0"
set r_compress "0"
set r_screenshotFormat "jpg"
set con_fontSize "14"
set vid_mode "8"
set vid_resizable "3"
set vid_display "-1"
set vid_fullscreen "0"
set vid_last "1777,889,83,100,0"

Hope this helps

**Comment by Openarl** [comment]
Hmm. There's nothing out of order in the config file, but that was a bit of a long shot anyway. Does the crash still occur if the window was on the second monitor, but not maximised? What about if the window is spread across both monitors?

**Comment by Cmpukahi** [comment]
Yes, application crashes for both of those situations. Something I noticed was that if I start to drag the window from the main monitor, the application will resize itself. If I close it in that state, the program will start just fine on the main monitor. If I try to resize it myself on the main monitor to something bigger, it crashes. Custom window size also crashes on second monitor. I wonder if it's a resizing issue instead of a monitor issue.

Crash occurs before any of the UI shows up. On a fresh wipe of the settings, the program does not take very much time to load up. After resizing/moving the window, once the initialization screen goes away, the main window opens and freezes as a white screen.

**Comment by xumoer** [comment]
I also have this problem. Deleting the line `set vid_last` seems to fix the problem. I wonder if there is some out of bounds issue or the variable storing the window location can't recognize negatives?

**Comment by Openarl** [comment]
I've spent some time on this, and I believe I've found a workaround for this issue by changing the way the program initialises its window and renderer. The fix will be deployed in the next update (1.4.64), and I'm looking forward to hearing from anyone still experiencing the issue to confirm that the fix has worked.

**Comment by Cmpukahi** [comment]
Sorry for the late reply. I just tried this out again and I'm receiving a black window when starting up the program. I noticed that some other people were reporting a black screen as well recently. I also tried a fresh download and no luck. Just to be clear, my program freezes during this black window and I have to force close it and wipe all the settings in the launch file.

Not sure if others have been having better results.

**Comment by jkluch** [comment]
I'm on the latest build 1.4.74 and the application won't start on my secondary screens, just primary.  The green startup script shows, but then the main window stays black and freezes.

**Comment by Cajden** [comment]
I have the same problem as jkluch, was there ever a solution found for it?
Tried reinstalling, resetting at various displays. And it just comes back to the black screen and freeze after the green startup.

**Comment by Ryukenwulf** [comment]
Same problem as well tried so many things its hard to list them all, works fine on my laptop but on my main PC it use to work but just stopped after the recent geforce driver update, not sure if that was related?

**Comment by jkluch** [comment]
While really frustrating I don't think this is an easy problem to debug for @Openarl since he doesn't have it himself. I'm still seeing the issue too even with reinstalled graphic drivers but at least there is a workaround if you start the exe from your primary monitor and then move it to the monitor of your choosing after it starts. I've zeroed out my Launch.cfg and marked it as read only so I don't have to worry about zeroing it out everytime I use PoB. I'll try running PoB in a development environment later and see if that provides any leads.

**Comment by Dunkdk** [comment]
I'll try the workaround, thanks! 

The only reason I posted was out of curiousity tbh. saw the last comment was from 28 March, and maybe the issue was fixed back then and forgot to put this as solved etc. I just had to make sure :)
